0100-private-projects.md

```diff
@@ -0,0 +1,36 @@
+- Feature Name: Private Projects
+- Created: 2020-02-10
+- Associated Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1602490
+
+# Summary
+
+Add an option to make a project "private", restricting who can see and interact with it.
+
+# Motivation
+
+It can be useful in some cases to be able to hide a project from users. For example, when a superuser creates a new project, they might want to make sure that everything is correctly set up before exposing it — running sync, verifying that it is enabled for the right locales, etc. In such a case, the project being not visible from non-superusers is enough. Once the superuser is done with configuration, they simply make the project public.
+
+For a different use-case, let's consider an agency working with several clients. They have a number of projects created on Pontoon, and they want to have specific localizers working on specific projects. It would be beneficial for them to able to restrict seeing a project to only a defined list of users. This requires being able to mark a project as private, and also having a configuration option to select a list of users who will be able to see, and interact, with that project. That would allow a clear separation of projects, reducing the risk of localizers translating the wrong set of projects.
+
+# Feature explanation
+
+When creating or editing a project, a superadmin has two new options:
+
+- a simple checkbox to make the project private or public;
+- a formset controlling which users can see the project when it is private.
```

Comment:
We should probably use the similar UX to locale selector in settings/admin or user selector in the permissions page. So two lists - one for "approved" (is "privileged" less confusing?) and the other for available users. There should also be an option to copy users from another project (same as we have for copying locales), because the list of privileged users is likely to repeat if I understand the use case correctly.

Reply:
Along with the option to copy users from another projects, Can we also have the ability to form a "user group" and allow copying from a user group. In fact, users from another project can be treated as a "user group" as well. I think this will add more flexibility to how a user is approved for a project.

Reply:
Groups would be ideal. They also require much more work than (reusing) the copy function.

Reply:
Groups would indeed be great, but I guess that's another feature entirely? 

@mathjazz regarding the two lists, I assume one of them has the full list of all users in the instance? That would not work in our case, as we have several thousand users. The list would be too long to load and near unusable.

Copy from a project sounds good though, and important enough.

Reply:
Trying the permissions page, your assumption seems correct - it gets pretty slow. We should however consolidate the selector UX with the permissions page, because this is basically the same thing.